### ISSUE TITLE
`systemd-tmpfiles-setup.service` can skip running in the real root

### ISSUE THREAD

**Issue by dustymabe** [systemd/systemd#23745]
**Descritpion**

In the Fedora CoreOS CI we have noticed a problem where `systemd-tmpfiles-setup.service` is not attempted to be run in the real root (it runs in the initramfs, but not after switching root). This causes all sorts of issues with systemd services that expect files/directories to exist.

Here is the journal log from a system that showed the issue: [journal.txt](https://github.com/coreos/fedora-coreos-tracker/files/8887583/journal.txt). These tests are being run in qemu/kvm virtual machines.

This problem is fairly consistent on s390x, which we just added to our Fedora CoreOS pipelines, but at it's core it is a race condition. We haven't seen the issue on other architectures currently.

Elements of this seem to overlap with the behavior described in the commit message of https://github.com/systemd/systemd/commit/2525682

This is also tracked in the Fedora CoreOS tracker at https://github.com/coreos/fedora-coreos-tracker/issues/1215

**systemd version the issue has been seen with**

```
$ systemctl --version
systemd 250 (v250.7-1.fc36)
+PAM +AUDIT +SELINUX -APPARMOR +IMA +SMACK +SECCOMP +GCRYPT +GNUTLS +OPENSSL +ACL +BLKID +CURL +ELFUTILS +FIDO2 +IDN2 -IDN -IPTC +KMOD +LIBCRYPTSETUP +LIBFDISK +PCRE2 +PWQUALITY +P11KIT +QRENCODE +BZIP2 +LZ4 +XZ +ZLIB +ZSTD +BPF_FRAMEWORK +XKBCOMMON +UTMP +SYSVINIT default-hierarchy=unified
```

**Used distribution**
Fedora CoreOS

```
$ rpm -q systemd
systemd-250.7-1.fc36.s390x
```

**Linux kernel version used** (`uname -a`)

```
[core@qemu0 ~]$ uname -a 
Linux qemu0 5.17.13-300.fc36.s390x #1 SMP Mon Jun 6 13:59:07 UTC 2022 s390x s390x s390x GNU/Linux
```

**CPU architecture issue was seen on**
- s390x

**Expected behaviour you didn't see**

`systemd-tmpfiles-setup.service` running in the real root and creating tmpfiles.

The following output taken from a machine not showing the problem:

```
$ sudo journalctl -p info UNIT=systemd-tmpfiles-setup.service + CODE_LINE=2049
Jun 13 02:53:29 localhost systemd[1]: Starting systemd-tmpfiles-setup.service - Create Volatile Files and Directories...
Jun 13 02:53:29 localhost systemd[1]: Finished systemd-tmpfiles-setup.service - Create Volatile Files and Directories.
Jun 13 02:53:37 qemu0 systemd[1]: systemd-tmpfiles-setup.service: Deactivated successfully.
Jun 13 02:53:37 qemu0 systemd[1]: Stopped systemd-tmpfiles-setup.service - Create Volatile Files and Directories.
Jun 13 02:53:37 qemu0 systemd[1]: Switching root.
Jun 13 02:53:39 qemu0 systemd[1]: Starting systemd-tmpfiles-setup.service - Create Volatile Files and Directories...
Jun 13 02:53:39 qemu0 systemd[1]: Finished systemd-tmpfiles-setup.service - Create Volatile Files and Directories.
```

**Unexpected behaviour you saw**

`systemd-tmpfiles-setup.service` not running in the real root.

The following output taken from a machine showing the problem:

```
$ sudo journalctl -p info UNIT=systemd-tmpfiles-setup.service + CODE_LINE=2049
Jun 13 02:24:32 localhost systemd[1]: Starting systemd-tmpfiles-setup.service - Create Volatile Files and Directories...
Jun 13 02:24:32 localhost systemd[1]: Finished systemd-tmpfiles-setup.service - Create Volatile Files and Directories.
Jun 13 02:24:40 qemu0 systemd[1]: Switching root.
```

**Steps to reproduce the problem**

I don't have a solid set of steps to reproduce the problem, but I am able to reproduce the problem consistently in the Fedora CoreOS CI system, which we can work together on debugging and testing theories for how to solve the problem.

**Comment by yuwata** [comment]
- All s390x machine has the issue? Or some?
- Is systemd-tmpfiles-setup.service is enabled? Please show the result of `systemctl status systemd-tmpfiles-setup.service`.
- Also show `systemctl cat systemd-tmpfiles-setup.service`.

**Comment by yuwata** [comment]
The service is by default statically wanted by sysinit.target (that is, symlink to the service is in /usr/lib/systemd/system/sysinit.target.wants). Is it true on your system?

**Comment by dustymabe** [comment]
Yes.

```
[core@qemu0 ~]$ ls -l /usr/lib/systemd/system/sysinit.target.wants/systemd-tmpfiles-setup.service 
lrwxrwxrwx. 2 root root 33 Jun 11 03:01 /usr/lib/systemd/system/sysinit.target.wants/systemd-tmpfiles-setup.service -> ../systemd-tmpfiles-setup.service
```

I don't know if thiw was clear before, but I can boot the exact same image (fresh) 10 times and reproduce this issue a percentage (maybe 50%) of the time. The image clearly has the correct setup because it works some of the time. Because it works some of the time it appears to me to be a race condition.

**Comment by dustymabe** [comment]
and to show it's not a broken symlink:

```
[core@qemu0 ~]$ 
[core@qemu0 ~]$ cat /usr/lib/systemd/system/sysinit.target.wants/systemd-tmpfiles-setup.service
#  SPDX-License-Identifier: LGPL-2.1-or-later
#
#  This file is part of systemd.
#
#  systemd is free software; you can redistribute it and/or modify it
#  under the terms of the GNU Lesser General Public License as published by
#  the Free Software Foundation; either version 2.1 of the License, or
#  (at your option) any later version.

[Unit]
Description=Create Volatile Files and Directories
Documentation=man:tmpfiles.d(5) man:systemd-tmpfiles(8)
DefaultDependencies=no
Conflicts=shutdown.target initrd-switch-root.service
After=local-fs.target systemd-sysusers.service systemd-journald.service
Before=sysinit.target shutdown.target
RefuseManualStop=yes

[Service]
Type=oneshot
RemainAfterExit=yes
ExecStart=systemd-tmpfiles --create --remove --boot --exclude-prefix=/dev
SuccessExitStatus=DATAERR CANTCREAT
```

**Comment by dustymabe** [comment]
Any more information I can provide? Would be nice to find the root cause of this problem.

**Comment by LakshmiRavichandran1** [comment]
are there more details needed from s390x on this?

**Comment by poettering** [comment]
please provide debug logs of the boot, otherwise this is not actionable

**Comment by poettering** [comment]
https://freedesktop.org/wiki/Software/systemd/Debugging/#diagnosingbootproblems

**Comment by poettering** [comment]
if you turn off selinux and all that mess, does it work then?

**Comment by dustymabe** [comment]
> please provide debug logs of the boot, otherwise this is not actionable

Here is a boot with `systemd.log_level=debug`: [journal-debug.txt](https://github.com/systemd/systemd/files/9067469/journal-debug.txt)



> if you turn off selinux and all that mess, does it work then?

With `enforcing=0` I still see the problem. The `journal.txt` linked from the original description above has `enforcing=0`.

**Comment by dustymabe** [comment]
any updates on this?

**Comment by dustymabe** [comment]
Hi @poettering, @keszybz any chance we could set up some time to debug this (through a screenshare?)

**Comment by poettering** [comment]
Can you provide the complete debug logs please? i.e add log_buf_len=8M or so on the kernel cmdline so that kmsg doesn't flush messages

**Comment by dustymabe** [comment]
Unfortunately it appears that setting `log_buf_len=8M` throws off the timing enough that the issue doesn't occur. Any other ideas?

I am able now to reproduce in `rawhide`, which I previously wasn't able to before. This is with `systemd-251.4-53.fc38`, which leads me to think the problem probably isn't fixed upstream.

**Comment by poettering** [comment]
> Unfortunately it appears that setting `log_buf_len=8M` throws off the timing enough that the issue doesn't occur. Any other ideas?

Try a couple of times with varying buffer sizes. Key is to get the logs from the transition, where systemd makes it mind up what to start.

You could also add a small service to the initrd, that enables debug output via "systemd-analyze log-level debug" just before the transition and then boot up without debug messages. That way hopefully the logging won't be so invasive and the race can still be triggered.

**Comment by dustymabe** [comment]
Ok I managed to get a log with `log_buf_len=512k`. I tried 1M to 8M and they all seemed to make it where it wasn't easy to reproduce.

I can confirm the problem did happen on this system because we notice `systemd-tmpfiles-setup.service` definitely doesn't run after the switch root:

```
[core@qemu0 ~]$ sudo journalctl -p info UNIT=systemd-tmpfiles-setup.service + _SYSTEMD_UNIT=initrd-switch-root.service | cat
Sep 02 02:40:54 localhost systemd[1]: Starting systemd-tmpfiles-setup.service - Create Volatile Files and Directories...
Sep 02 02:40:54 localhost systemd[1]: Finished systemd-tmpfiles-setup.service - Create Volatile Files and Directories.
Sep 02 02:41:03 qemu0 systemctl[1169]: Bus n/a: changing state UNSET → OPENING
Sep 02 02:41:03 qemu0 systemctl[1169]: sd-bus: starting bus by connecting to /run/systemd/private...
Sep 02 02:41:03 qemu0 systemctl[1169]: Bus n/a: changing state OPENING → AUTHENTICATING
Sep 02 02:41:03 qemu0 systemctl[1169]: Switching root - root: /sysroot; init: n/a
Sep 02 02:41:03 qemu0 systemctl[1169]: Bus n/a: changing state AUTHENTICATING → RUNNING
Sep 02 02:41:03 qemu0 systemctl[1169]: Sent message type=method_call sender=n/a destination=org.freedesktop.systemd1 path=/org/freedesktop/systemd1 interface=org.freedesktop.systemd1.Manager member=SwitchRoot cookie=1 reply_cookie=0 signature=ss error-name=n/a error-message=n/a
Sep 02 02:41:03 qemu0 systemctl[1169]: Got message type=method_return sender=org.freedesktop.systemd1 destination=n/a path=n/a interface=n/a member=n/a cookie=1 reply_cookie=1 signature=n/a error-name=n/a error-message=n/a
Sep 02 02:41:03 qemu0 systemctl[1169]: Bus n/a: changing state RUNNING → CLOSED
```

full log: [journalctl_log_buf_len_512k.txt](https://github.com/systemd/systemd/files/9474577/journalctl_log_buf_len_512k.txt)

**Comment by poettering** [comment]
Hmm, the transaction shown does not mention systemd-tmpfiles.service at all. Are you sure it's properly linked up in /usr/lib/systemd/system/sysinit.target.wants/ in such systems?

very strange... if you turn off selinux fully (i.e. selinux=0 audit=0) does that change anything?

**Comment by dustymabe** [comment]
> Hmm, the transaction shown does not mention systemd-tmpfiles.service at all. Are you sure it's properly linked up in /usr/lib/systemd/system/sysinit.target.wants/ in such systems?

I think I answered this question already in https://github.com/systemd/systemd/issues/23745#issuecomment-1158814225 and https://github.com/systemd/systemd/issues/23745#issuecomment-1158815389

> 
> very strange... if you turn off selinux fully (i.e. selinux=0 audit=0) does that change anything?

Unfortunately I don't think I can completely disable selinux in our setup because of https://github.com/coreos/rpm-ostree/issues/971

Any other ideas?

**Comment by poettering** [comment]
> Unfortunately I don't think I can completely disable selinux in our setup because of [coreos/rpm-ostree#971](https://github.com/coreos/rpm-ostree/issues/971)

I am sorry, what? ostree requires selinux? wtf? why? just why? (the issue doesn't really explain why...)

can you provide "systemctl show systemd-tmpfiles-setup.service" when this happens?

a wild guess, but is getdents64() reliable on your arch? this kinda feels as if it might not be? i.e. continuation getdents64() might be hosed?

**Comment by dustymabe** [comment]
> > Unfortunately I don't think I can completely disable selinux in our setup because of [coreos/rpm-ostree#971](https://github.com/coreos/rpm-ostree/issues/971)
> 
> I am sorry, what? ostree requires selinux? wtf? why? just why? (the issue doesn't really explain why...)

`rpm-ostree`, not `ostree`, but yes. I don't know why. We should comment/discuss over in that ticket if we want more clarity.

> 
> can you provide "systemctl show systemd-tmpfiles-setup.service" when this happens?

<details>

```
[core@qemu0 ~]$ sudo systemctl show systemd-tmpfiles-setup.service | cat
Type=oneshot
ExitType=main
Restart=no
NotifyAccess=none
RestartUSec=100ms
TimeoutStartUSec=infinity
TimeoutStopUSec=1min 30s
TimeoutAbortUSec=1min 30s
TimeoutStartFailureMode=terminate
TimeoutStopFailureMode=terminate
RuntimeMaxUSec=infinity
RuntimeRandomizedExtraUSec=0
WatchdogUSec=0
WatchdogTimestampMonotonic=0
RootDirectoryStartOnly=no
RemainAfterExit=yes
GuessMainPID=yes
SuccessExitStatus=65 73
MainPID=0
ControlPID=0
FileDescriptorStoreMax=0
NFileDescriptorStore=0
StatusErrno=0
Result=success
ReloadResult=success
CleanResult=success
UID=[not set]
GID=[not set]
NRestarts=0
OOMPolicy=stop
ExecMainStartTimestamp=Fri 2022-09-02 18:05:10 UTC
ExecMainStartTimestampMonotonic=1655557
ExecMainExitTimestamp=Fri 2022-09-02 18:05:10 UTC
ExecMainExitTimestampMonotonic=1670658
ExecMainPID=222
ExecMainCode=1
ExecMainStatus=0
ExecStart={ path=systemd-tmpfiles ; argv[]=systemd-tmpfiles --create --remove --boot --exclude-prefix=/dev ; ignore_errors=no ; start_time=[n/a] ; stop_time=[n/a] ; pid=0 ; code=(null) ; status=0/0 }
ExecStartEx={ path=systemd-tmpfiles ; argv[]=systemd-tmpfiles --create --remove --boot --exclude-prefix=/dev ; flags= ; start_time=[n/a] ; stop_time=[n/a] ; pid=0 ; code=(null) ; status=0/0 }
Slice=system.slice
ControlGroupId=0
MemoryCurrent=[not set]
MemoryAvailable=infinity
CPUUsageNSec=5344000
TasksCurrent=[not set]
IPIngressBytes=[no data]
IPIngressPackets=[no data]
IPEgressBytes=[no data]
IPEgressPackets=[no data]
IOReadBytes=18446744073709551615
IOReadOperations=18446744073709551615
IOWriteBytes=18446744073709551615
IOWriteOperations=18446744073709551615
Delegate=no
CPUAccounting=yes
CPUWeight=[not set]
StartupCPUWeight=[not set]
CPUShares=[not set]
StartupCPUShares=[not set]
CPUQuotaPerSecUSec=infinity
CPUQuotaPeriodUSec=infinity
IOAccounting=no
IOWeight=[not set]
StartupIOWeight=[not set]
BlockIOAccounting=no
BlockIOWeight=[not set]
StartupBlockIOWeight=[not set]
MemoryAccounting=yes
DefaultMemoryLow=0
DefaultMemoryMin=0
MemoryMin=0
MemoryLow=0
MemoryHigh=infinity
MemoryMax=infinity
MemorySwapMax=infinity
MemoryLimit=infinity
DevicePolicy=auto
TasksAccounting=yes
TasksMax=2314
IPAccounting=no
ManagedOOMSwap=auto
ManagedOOMMemoryPressure=auto
ManagedOOMMemoryPressureLimit=0
ManagedOOMPreference=none
UMask=0022
LimitCPU=infinity
LimitCPUSoft=infinity
LimitFSIZE=infinity
LimitFSIZESoft=infinity
LimitDATA=infinity
LimitDATASoft=infinity
LimitSTACK=infinity
LimitSTACKSoft=8388608
LimitCORE=infinity
LimitCORESoft=infinity
LimitRSS=infinity
LimitRSSSoft=infinity
LimitNOFILE=524288
LimitNOFILESoft=1024
LimitAS=infinity
LimitASSoft=infinity
LimitNPROC=7716
LimitNPROCSoft=7716
LimitMEMLOCK=8388608
LimitMEMLOCKSoft=8388608
LimitLOCKS=infinity
LimitLOCKSSoft=infinity
LimitSIGPENDING=7716
LimitSIGPENDINGSoft=7716
LimitMSGQUEUE=819200
LimitMSGQUEUESoft=819200
LimitNICE=0
LimitNICESoft=0
LimitRTPRIO=0
LimitRTPRIOSoft=0
LimitRTTIME=infinity
LimitRTTIMESoft=infinity
OOMScoreAdjust=0
CoredumpFilter=0x33
Nice=0
IOSchedulingClass=2
IOSchedulingPriority=4
CPUSchedulingPolicy=0
CPUSchedulingPriority=0
CPUAffinityFromNUMA=no
NUMAPolicy=n/a
TimerSlackNSec=50000
CPUSchedulingResetOnFork=no
NonBlocking=no
StandardInput=null
StandardOutput=journal
StandardError=inherit
TTYReset=no
TTYVHangup=no
TTYVTDisallocate=no
SyslogPriority=30
SyslogLevelPrefix=yes
SyslogLevel=6
SyslogFacility=3
LogLevelMax=-1
LogRateLimitIntervalUSec=0
LogRateLimitBurst=0
SecureBits=0
CapabilityBoundingSet=cap_chown cap_dac_override cap_dac_read_search cap_fowner cap_fsetid cap_kill cap_setgid cap_setuid cap_setpcap cap_linux_immutable cap_net_bind_service cap_net_broadcast cap_net_admin cap_net_raw cap_ipc_lock cap_ipc_owner cap_sys_module cap_sys_rawio cap_sys_chroot cap_sys_ptrace cap_sys_pacct cap_sys_admin cap_sys_boot cap_sys_nice cap_sys_resource cap_sys_time cap_sys_tty_config cap_mknod cap_lease cap_audit_write cap_audit_control cap_setfcap cap_mac_override cap_mac_admin cap_syslog cap_wake_alarm cap_block_suspend cap_audit_read cap_perfmon cap_bpf cap_checkpoint_restore
DynamicUser=no
RemoveIPC=no
PrivateTmp=no
PrivateDevices=no
ProtectClock=no
ProtectKernelTunables=no
ProtectKernelModules=no
ProtectKernelLogs=no
ProtectControlGroups=no
PrivateNetwork=no
PrivateUsers=no
PrivateMounts=no
PrivateIPC=no
ProtectHome=no
ProtectSystem=no
SameProcessGroup=no
UtmpMode=init
IgnoreSIGPIPE=yes
NoNewPrivileges=no
SystemCallErrorNumber=2147483646
LockPersonality=no
RuntimeDirectoryPreserve=no
RuntimeDirectoryMode=0755
StateDirectoryMode=0755
CacheDirectoryMode=0755
LogsDirectoryMode=0755
ConfigurationDirectoryMode=0755
TimeoutCleanUSec=infinity
MemoryDenyWriteExecute=no
RestrictRealtime=no
RestrictSUIDSGID=no
RestrictNamespaces=no
MountAPIVFS=no
KeyringMode=private
ProtectProc=default
ProcSubset=all
ProtectHostname=no
KillMode=control-group
KillSignal=15
RestartKillSignal=15
FinalKillSignal=9
SendSIGKILL=yes
SendSIGHUP=no
WatchdogSignal=6
Id=systemd-tmpfiles-setup.service
Names=systemd-tmpfiles-setup.service
Requires=system.slice
WantedBy=sysinit.target
Conflicts=shutdown.target initrd-switch-root.service
Before=rpm-ostreed.service systemd-resolved.service systemd-update-utmp.service systemd-journal-catalog-update.service systemd-timesyncd.service sysinit.target rpm-ostree-countme.service chronyd.service logrotate.service dbus-broker.service systemd-logind.service shutdown.target
After=systemd-journald.socket system.slice ostree-remount.service systemd-sysusers.service systemd-journald.service systemd-journal-flush.service local-fs.target
Documentation="man:tmpfiles.d(5)" "man:systemd-tmpfiles(8)"
Description=Create Volatile Files and Directories
LoadState=loaded
ActiveState=active
FreezerState=running
SubState=exited
FragmentPath=/usr/lib/systemd/system/systemd-tmpfiles-setup.service
UnitFileState=static
UnitFilePreset=disabled
StateChangeTimestamp=Fri 2022-09-02 18:05:10 UTC
StateChangeTimestampMonotonic=1670715
InactiveExitTimestamp=Fri 2022-09-02 18:05:10 UTC
InactiveExitTimestampMonotonic=1655584
ActiveEnterTimestamp=Fri 2022-09-02 18:05:10 UTC
ActiveEnterTimestampMonotonic=1670715
ActiveExitTimestampMonotonic=0
InactiveEnterTimestampMonotonic=0
CanStart=yes
CanStop=no
CanReload=no
CanIsolate=no
CanFreeze=yes
StopWhenUnneeded=no
RefuseManualStart=no
RefuseManualStop=yes
AllowIsolate=no
DefaultDependencies=no
OnSuccessJobMode=fail
OnFailureJobMode=replace
IgnoreOnIsolate=no
NeedDaemonReload=no
JobTimeoutUSec=infinity
JobRunningTimeoutUSec=infinity
JobTimeoutAction=none
ConditionResult=yes
AssertResult=yes
ConditionTimestamp=Fri 2022-09-02 18:05:10 UTC
ConditionTimestampMonotonic=1655141
AssertTimestamp=Fri 2022-09-02 18:05:10 UTC
AssertTimestampMonotonic=1655142
Transient=no
Perpetual=no
StartLimitIntervalUSec=10s
StartLimitBurst=5
StartLimitAction=none
FailureAction=none
SuccessAction=none
InvocationID=c663acc3d5c3472c999e5f0a3ebde247
CollectMode=inactive
```

</details>


> 
> a wild guess, but is getdents64() reliable on your arch? this kinda feels as if it might not be? i.e. continuation getdents64() might be hosed?


Maybe not? This only happens on `s390x` so maybe there is an issue with that platform. Do you have a good theory on how to investigate that?

**Comment by dustymabe** [comment]
Hey @poettering - any insight you could gain from the `systemctl show` output? Or maybe ideas on how we can determine if the `s390x` platform is buggy in some way in Fedora? If there's a small program we can compile and run to check then I can try to do that.

**Comment by poettering** [comment]
Sorry, got nothing. You might need to start with some gdb or printf() based debugging, figuring out why the service is not pulled in even if it supposedly has WantedBy=sysinit.target...

**Comment by keszybz** [comment]
```
Sep 02 02:40:54 localhost systemd-sysusers[216]: Creating group 'nobody' with GID 65534.
Sep 02 02:40:54 localhost systemd-sysusers[216]: Creating group 'root' with GID 999.

Sep 02 02:40:56 localhost dbus-broker-launch[726]: Looking up NSS user entry for 'dbus'...
Sep 02 02:40:56 localhost dbus-broker-launch[726]: NSS returned no entry for 'dbus'
Sep 02 02:40:56 localhost dbus-broker-launch[726]: Invalid user-name in /usr/share/dbus-1/system.conf +18: <user>dbus</user>

Sep 02 02:40:58 qemu0 systemd-tmpfiles[954]: Failed to parse ACL "default:group:tss:rwx": Invalid argument. Ignoring
```
Hmm. Seems there's something wrong with the user/group tables in the initrd. I'm not sure what effect having group `root` with a wrong UUID can have. Maybe nothing, except some confusion for users? Some other users/groups are also missing.

But we're also missing many messages. So maybe there *was* some error, but just don't know about it…
```
Sep 02 02:40:54 localhost systemd-journald[213]: Missed 232 kernel messages
```

**Comment by cgwalters** [comment]
> Unfortunately I don't think I can completely disable selinux in our setup because of https://github.com/coreos/rpm-ostree/issues/971

Yes, you can boot with `selinux=0` here.

**Comment by poettering** [comment]
A group "root" with UID 999? I didn't even see that! That's definitely a bug somewhere. Do you see that on all archs? And on all runs?

**Comment by poettering** [comment]
So, what happens if you add the following drop-in to your initrd:

```
#/etc/systemd/system/systemd-tmpfiles.service.d/order-fix.conf
Before=initrd-switch-root.target
```

**Comment by dustymabe** [comment]
I added `Before=initrd-switch-root.target` to `systemd-tmpfiles-setup.service`, which yielded:

```
$ systemctl cat systemd-tmpfiles-setup.service
# /usr/lib/systemd/system/systemd-tmpfiles-setup.service
#  SPDX-License-Identifier: LGPL-2.1-or-later
#
#  This file is part of systemd.
#
#  systemd is free software; you can redistribute it and/or modify it
#  under the terms of the GNU Lesser General Public License as published by
#  the Free Software Foundation; either version 2.1 of the License, or
#  (at your option) any later version.

[Unit]
Description=Create Volatile Files and Directories
Documentation=man:tmpfiles.d(5) man:systemd-tmpfiles(8)
DefaultDependencies=no
Conflicts=shutdown.target initrd-switch-root.service
After=local-fs.target systemd-sysusers.service systemd-journald.service
Before=sysinit.target shutdown.target initrd-switch-root.target
RefuseManualStop=yes

[Service]
Type=oneshot
RemainAfterExit=yes
ExecStart=systemd-tmpfiles --create --remove --boot --exclude-prefix=/dev
SuccessExitStatus=DATAERR CANTCREAT
```

and I've ran the test 20 times and am not able to reproduce the issue. I can run it some more, but it looks like that particular change does help.

**Comment by dustymabe** [comment]
For the `GID` issue I opened https://github.com/coreos/fedora-coreos-tracker/issues/1297 so we can investigate that further.

**Comment by cgwalters** [comment]
On Sat, Sep 17, 2022, at 7:27 AM, Zbigniew Jędrzejewski-Szmek wrote:
> Closed #23745 <https://github.com/systemd/systemd/issues/23745> as 
> completed via #24670 <https://github.com/systemd/systemd/pull/24670>.

Thanks for digging into this!
